clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -94,6 +95,85 @@ def SInt16 : SInt<16>;
 def SInt32 : SInt<32>;
 def SInt64 : SInt<64>;
 
+//===----------------------------------------------------------------------===//
+// FloatType
+//===----------------------------------------------------------------------===//
+
+def CIR_FloatType : CIR_Type<"Float", "float",
+    [DeclareTypeInterfaceMethods<DataLayoutTypeInterface>]> {
+  let summary = "CIR floating-point type";
+  let description = [{
+    CIR type that represents C/C++ floating-point types, including `float`, \
+    `double`, and `long double`.
+  }];
+  let parameters = (ins "::mlir::cir::FloatType::Format":$format);
+
+  let builders = [
+    TypeBuilder<(ins CArg<"::mlir::cir::FloatType::Format">:$format)>
+  ];
+
+  let hasCustomAssemblyFormat = 1;
+  let genVerifyDecl = 1;
+
+  let extraClassDeclaration = [{
+    // Floating-point format.
+    enum Format : uint32_t {
+      FF_Binary16,   // IEEE-754 binary16 format
+      FF_Binary32,   // IEEE-754 binary32 format
+      FF_Binary64,   // IEEE-754 binary64 format
+      FF_Binary64E,  // IEEE-754 binary64-extended format
+      FF_Binary128,  // IEEE-754 binary128 format
+      FF_BFloat16,   // Brain floating format
+    };
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary16 format.
+    bool isBinary16() const { return getFormat() == FF_Binary16; }
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary32 format. This typically corresponds to `float` type.
+    bool isBinary32() const { return getFormat() == FF_Binary32; }
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary64 format. This typically corresponds to `double` type.
+    bool isBinary64() const { return getFormat() == FF_Binary64; }
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary64-extended format. This may correspond to `long double` type
+    /// depending on the implementation.
+    bool isBinary64Extended() const { return getFormat() == FF_Binary64E; }
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// binary128 format. This may correspond to `long double` type depending
+    /// on the implementation.
+    bool isBinary128() const { return getFormat() == FF_Binary128; }
+
+    /// Return true if values of this floating-point type is in IEEE-754
+    /// bfloat16 format.
+    bool isBFloat16() const { return getFormat() == FF_BFloat16; }
```

Comment:
`bfloat16` is not an IEEE-754 format.  Please update the comment.

Reply:
Updated.